src/ide/completion/path.rs

```diff
@@ -125,12 +137,25 @@ pub fn path_suffix_completions<'db>(
                 _ => None,
             };
 
+            let snippet_text = get_snippet_text(
+                last_segment,
+                is_declarative_macro,
+                struct_initialization_snippet_text.clone(),
+            );
+
             Some(CompletionItemOrderable {
                 item: CompletionItem {
-                    label: last_segment.to_string(),
-                    insert_text: struct_initialization_text.clone(),
-                    insert_text_format: struct_initialization_text
-                        .map(|_| InsertTextFormat::SNIPPET),
+                    label: if is_declarative_macro {
+                        format!("{}!", last_segment)
+                    } else {
+                        last_segment.to_string()
+                    },
+                    insert_text: get_snippet_text(
+                        last_segment,
+                        is_declarative_macro,
+                        struct_initialization_snippet_text.clone(),
+                    ),
```

Comment:
Those two could be calculated in tandem i think, making for simpler logic to track

Reply:
it's basically the same if (only thing the function has) - why not calculate it in the same branch? if not, let's have separate function that does the same thing, or inline it as well. now it's mixed and it seems weird